### DISCUSSION TITLE
DNSCrypt-proxy + VPN + socks5

### DISCUSSION THREAD

**Discussion by Leonis87** [DNSCrypt/dnscrypt-proxy#2062] [Q&A]
Hi,

I have question about using dnscrypt-proxy + vpn + socks5. I'm using mullvad VPN and it has proxy socks5 option. I know how to use VPN with dnscrypt-proxy, just add custom server 127.0.0.1 in Mullvad app and it shows DNS leak which is correct. But when I use mullvad proxy socks5 (openVPN or Wireguard server) in firefox browser, DNS leak test show only mullvad DNS server, and it's not using dnscrypt proxy for some reason. Is it alright ? Am I doing something wrong ?

I'm using VPN with dnscrypt quite often because my ISP is doing DPI. I don't use VPN for banking, social media, government applications etc... But I don't like my ISP see everything. I didn't found much answers on google so I came here maybe somebody can clarify for me...

**Comment by lifenjoiner** [comment]
VPN uses a local dnscrypt-proxy? I doubt it:
VPN may use a local DNS to solve a VPN server name. But, in a VPN environment, it should use it's own DNS which is reachable on its own network, or a LAN server name may not be solved.
For SOCKS5, if the browser `CONNECT`s a server by its name, the name would be solved by the DNS in the VPN; if it `CONNECT`s a server by its IP, the browser solves the name locally first.

Will mullvad VPN make an exception? It depends on how it exactly works.

**Comment by Leonis87** [comment]
That's true VPN should use it's own DNS server. I'm aware that using VPN app with added custom server in this case DNScrypt-proxy can harm privacy. But I trust Mullvad and I have anonymized DNScrypt proxy configuration in toml.file. I'm doing it this way because as I mentioned above I'm using VPN quite often and I don't want dnscrypt service-stop all the time and then flush/dns, ipconfig/renew, or even restart computer.

Mullvad use proxy located on IP 10.8.0.1 (OpenVPN) or 10.64.0.1 (WireGuard), port 1080. I'm using wireguard in firefox network setting which is 10.64.0.1, port 1080.
I don't know how Mullvad app exactly works. I'm gonna send email to Mullvad and ask them about DNScrypt+VPN+socks5.
